### ISSUE TITLE
Doubt about dataloader：data and labels?

### ISSUE THREAD

**Issue by boreas-l** [BGU-CS-VIL/DeepDPM#12]
Thankyou for your  nice contribution!  
I want to ask some problems...  
and in issue<https://github.com/BGU-CS-VIL/DeepDPM/issues/6>, you said ''the DeepDPM is an unsupervised method, labels are not used in any training stage (we use them in our code only for evaluation, but they are not used in training"  
but in the DeepDPM.py，the dataloader all will return datas and labels？I don't understand，I need your's  help.Thanks!  
Also, in path "./pretrained_embeedings/MoCO/.."，the traincodes.pt and labels.pt is mean what?

**Comment by MengXinChengXuYuan** [comment]
In my understanding, DeepDPM don't using any supervised information, which means it don't need labels to calculate losses. Labels are only used to eval the acc, etc, of the training cluster. If you don't have the labels of your data, simply generate random ints as label, but which these fake labels, acc nmi etc will means nothing

**Comment by meitarronen** [comment]
Thank you again @MengXinChengXuYuan!
Indeed, we are using the labels **only** to evaluate our clustering performance. As you can see, the labels are not used as input for the DeepDPM net.
As the method does not require labels, for a "real" unsupervised dataset, you could simply remove the evaluation (and some of the plotting) functions, and replace them with unsupervised criteria (e.g., silhouette score). Another option is, as @MengXinChengXuYuan suggested, you could generate mock labels and ignore the evaluation functions.

Regarding the files under "./pretrained_embeedings/MoCO/..": these are pretrained embeddings that we provide for easy reproduction of our code for each dataset (organized in different dirs). E.g., for ImageNet-50 you could see the embeddings that were generated using MoCO (unsupervised) and the corresponding labels, which are traincodes and labels respectively.

**Comment by boreas-l** [comment]
> In my understanding, DeepDPM don't using any supervised information, which means it don't need labels to calculate losses. Labels are only used to eval the acc, etc, of the training cluster. If you don't have the labels of your data, simply generate random ints as label, but with these fake labels, acc nmi etc will means nothing

Thankyou!! so nice!!
